### ISSUE TITLE
[Cron Monitoring] Getting error "Method Illuminate\Support\Str::replace does not exist"

### ISSUE THREAD

**Issue by fredericgboutin-yapla** [getsentry/sentry-laravel#761]
### How do you use Sentry?

Sentry SaaS (sentry.io)

### SDK version

3.7.3

### Steps to reproduce

I simply followed the instructions https://docs.sentry.io/platforms/php/guides/laravel/crons/ basically adding `->sentryMonitor()` calls to my scheduled commands.

### Expected result

Well, the whole thing ended up generating a Sentry event 😅 pointing to this code,

https://github.com/getsentry/sentry-laravel/blob/f8685aac8880701628bf24848b0c35f09b6888f3/src/Sentry/Laravel/Features/ConsoleIntegration.php#L177

It seems `Str::replace` was introduced in Laravel `8.41.0` https://github.com/laravel/framework/releases/tag/v8.41.0 but the current composer dependency requirements for this projet includes support for Laravel 6, which I'm shamefully using at the moment but doesn't provide an implementation for `Str::replace` 

### Actual result

As a workaround, I suppose I should explicitly provide a `monitorSlug` to avoid this error and make it work.

**Comment by fredericgboutin-yapla** [comment]
The workaround works, `->sentryMonitor('my-command')`

When doing it, it came to my attention that in Laravel you can define a `name` to schedule command with `->name('my-command')`.

I think it would be better if, instead of automatically creating a `slug` based on the command line itself which can get pretty ugly BTW, the code would first try to create it based on the name of the command when it is defined.

At moment I kind of duplicate the name of the slug with the name of my command anyway,

```php
$schedule->command('stupidly long and --with args that are --ugly > as hell')
    ->sentryMonitor('my-command')
    ->name('my-command')
    (...);
```

**Comment by stayallive** [comment]
Interesting. I must admit that I failed to check the replace method existed 🫣 We are going to fix that of course! Thanks for your report. 

You are right that having a slug set yourself is usually cleaner, but not always unless you indeed have a very long and ugly command, which is where you can define a name yourself if needed.

And using the name is not a bad idea there, honestly did not even knew you could do that... so gonna fix that too!

**Comment by stayallive** [comment]
So I've been thinking about the name, which is internally set as the description. I think that is the reason I never used it since description sounded like the command description. 

I was thinking of starting to use the description to generate the slug, however, that would be a backwards incompatible change and might generate new schedules if a description was set before this cahnge... so I don't think we can safely make that change anymore until the next major unfortunately. 

I did remove the `Str::replace()` usage ofcourse, see: #762.

**Comment by fredericgboutin-yapla** [comment]
But the feature is tagged as "beta". Personaly I wouldn't care to break the contract...

**Comment by stayallive** [comment]
@cleptric I'll leave you as the decider for this.

Long story short, there is anoter field we can use instead of the command name that might be set for scheduled commands that is probably prefered over a slug of the command, but doing so know might create duplicate monitors when users having `->sentryMonitor()` without a slug and without a proper command name so they fallback to the slugged command name but if they have a name/description set we would probably better slug that instead which would take precedense.

**Comment by fredericgboutin-yapla** [comment]
We went into production with Cron monitoring and when manually configuring some monitors in a non-laravel project, I saw this,

![image](https://github.com/getsentry/sentry-laravel/assets/91906601/5061006f-ad0f-4883-8fc2-55e17285ced1)

So, I think... if we want to be consistent here. We should use the Laravel schedule name as the Sentry Monitor name, when it exists. That's for sure.

Now about the slug, by default, yeah... it would kind of not bug me anymore if the slug is the Laravel schedule command as long as the Sentry Monitor name follows the Laravel schedule name.

Now, with the bug corrected I could simply

```php
$schedule->command('stupidly long and --with args that are --ugly > as hell')
    ->sentryMonitor()
    ->name('my-command')
    (...);
```

And that would be alright I think.

**Comment by cleptric** [comment]
I'm not in favour of making a change that could lead to users receiving a multitude of alters as the slug of their monitors changed.
We already support the use case of setting up a monitor in the UI first and adding the slug manually to a scheduled task.

Exposing the description during ingestion might be worth considering, but this discussion is better suited over at https://github.com/getsentry/sentry/discussions/42283, as it requires changes in Relay and consumers.

**Comment by cleptric** [comment]
This was fixed in [3.8.0](https://github.com/getsentry/sentry-laravel/releases/tag/3.8.0).
